napari/components/viewer_model.py

```diff
@@ -136,10 +136,35 @@ class ViewerModel(KeymapProvider, MousemapProvider, EventedModel):
     ----------
     window : Window
         Parent window.
+    camera: napari.Camera
+        The Napari camera object modeling the position and view.
+    cursor: napari.Cursor
+        The Napari cursor object containing the position and properties of the cursor.
+    grid: napari.Gridcanvas
+        The Napari gridcanvas allowing for the current implementation of a gridview of the canvas.
     layers : LayerList
         List of contained layers.
     dims : Dimensions
         Contains axes, indices, dimensions and sliders.
+    help: str
+        A help message of the viewer model
+    tooltip: napari.Tooltip
+        A tooltip showing extra information on the cursor
+    theme: str
+        Name of the Napari theme of the viewer
+    canvas_size: Tuple[int, int]
+        The canvas size following the Numpy convention of height x width
+    title: str
+        The title of the viewer model
+    overlays: napari.EventedDict[str, Overlay]
+        An EventedDict with as keys the string names of different napari overlays and as values the napari.Overlay
+        objects.
+    ctx: Mapping
+        Viewer object context mapping.
+    mouse_over_canvas: bool
+        Indicating whether the mouse cursor is on the viewer canvas.
+    layer_slicer: napari._Layer_Slicer
```

Comment:
Sometimes the way of documenting private attributes in general is confusing. For example in the QtViewer private attributes like docklayerlist do have a docstring. The argument that QtViewer is private so that would not matter here does not hold I think because the QtViewer can also be accessed publicly in order to embed in plugins. 

I can change it, but then I would also change it for the QtViewer docstring in order to be consistent. I am fine with inline comments. I will ask during the community meeting tomorrow.

Reply:
Agreed that the current state is confusing. And that the content of your additions is useful.

But I feel pretty strongly that private attributes shouldn't be documented alongside public ones because:

1\. It makes the docstring larger and thus harder to read for users/clients of the class that don't care about the implementation details.

2\.Encourages users/client of this class to use private attributes that could change or be removed in future versions.

As with [the numpydoc guide](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes) describing private methods, I'd be OK listing private attributes under `Notes`, though my real preference is outside of the docstring altogether.

Reply:
@andy-sweet we discussed this issue in the community meeting, and @psobolewskiPhD, @Czaki, and I agreed with @melonora that although you make a good point, we could/should fix this issue more globally, for example by writing a sphinx extension that removes any attributes starting with an underscore from the built docs. This should be (😅) relatively straightforward. In the meantime we were ok with leaving private attributes listed given that we are already doing that and that they are useful for other devs and even core devs who might not be familiar with this part of the code (🙋). @melonora will create a separate issue about private attributes in docstrings so we can keep track of where we want this particular aspect to land.

Reply:
I won't put up a fight in this PR, since overall it's great. However, I would also expect not to be blocked if I move some of these docs to the implementation in related PRs for similar reasons to the above (we already do that and they're still useful).

At a minimum, the attributes should have correct names - i.e. attributes with leading underscores in code should have leading underscores in the docstring. I'd also recommend grouping them and moving them to the bottom.

Reply:
For now I grouped private attributes below public attributes and they are indicated with leading underscore. Docstrings will be fixed with a separate PR and issue regarding this is created ([bf65dcf](https://github.com/napari/napari/pull/5432/commits/bf65dcf0dfea3a762cce4b2198c2d19bf23e152b).